### PR TITLE
[Testcase Refactoring] Decouple TestBaseDataScheduler for out-of-tree backends

### PR DESCRIPTION
**Pull request by CunYuawa**
## Summary

Decouple `test/ao/sparsity/test_data_scheduler.py` so it can be scheduled on any
registered backend (CPU / CUDA / privateuse1 accelerators such as NPU).

Changes:
- `test/ao/sparsity/test_data_scheduler.py`
  - Import `HardwareClassification` from `torch.testing._internal.common_utils`;
  - Annotate `TestBaseDataScheduler` with
    `hw_classification = HardwareClassification.GENERIC`: the class only exercises
    device-agnostic scheduler bookkeeping (constructor checks, step ordering,
    step counts, and `state_dict` round-trips) and does not depend on any
    particular accelerator, so it is marked as GENERIC.

No test semantics, case count, or execution order are changed.

## Test Plan

- CPU:
  - `python -m pytest test/ao/sparsity/test_data_scheduler.py -v`
    -> 4 passed (test_constructor / test_order_of_steps / test_step / test_state_dict)
  - `python -m pytest test/test_ao_sparsity.py -k DataScheduler -v`
    -> 4 passed
- CUDA: covered by upstream CI (CUDA jobs).

### CODE REVIEW

**Review by xuyun44 — APPROVED** [review]

**Review by orangeH25 — APPROVED** [review]

**Review by fffrog — APPROVED** [review]
SGTM

**Comment by CunYuawa** [comment]
Hi @jerryzh168 @fffrog, this PR is ready: all 156 CI checks passed, EasyCLA is signed, and there are no merge conflicts. Could you please help merge it when convenient? Happy to address any feedback. Thanks!

**Comment by CunYuawa** [comment]
Hi @albanD, just a gentle nudge — this small test-only refactoring PR is ready to merge: all 156 CI checks passed, EasyCLA is signed, and there are no merge conflicts. It already has approvals from @xuyun44, @orangeH25, and @fffrog. Could you please take a look and merge when convenient? Happy to address any feedback. Thanks!
